book/build-service.md

```diff
@@ -0,0 +1,158 @@
+
+#### PipelineRun selection
+
+The Build Service owns [BuildPipelineSelector CRD](https://redhat-appstudio.github.io/book/ref/build-service.html#buildpipelineselector) which defines which PipelineRun to select for the Component CR. By default global BuildPipelineSelector `build-pipeline-selector` in `build-service` namespace is used. BuildPipelineSelector CR contains selectors, the first matching selector is used for the Component CR. The list of selectors can be extended by creating BuildPipelineSelector CR with the same name as the Application CR in the user's namespace. This will ensure that it is applied to Component CRs under the corresponding Application CR. The list of selectors can be also extended for the whole user namespace by creating BuildPipelineSelector CR named `build-pipeline-selector` in the user's namespace.
+
+Selectors are processed in this order:
+- BuildPipelineSelector CR with the same name as Application CR in the user's namespace
+- BuildPipelineSelector CR named `build-pipeline-selector` in the user's namespace
+- BuildPipelineSelector CR named `build-pipeline-selector` in `build-service` namespace
+
+Example of BuildPipelineSelector:
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: BuildPipelineSelector
+metadata:
+  name: build-pipeline-selector
+  namespace: my-tenant
+spec:
+  selectors:
+  - name: my-component
+    pipelineRef:
+      bundle: quay.io/my-user/my-bundle:v1.0
+      name: my-bundle
+    pipelineParams:
+    - name: test
+      value: fbc
+    when:
+      language: java
+      componentName: my-component
+      projectType: springboot
+  - name: Label section
+    pipelineRef:
+      bundle: quay.io/my-user/my-bundle:v1.0
+      name: labelled
+    when:
+      labels:
+        mylabel: test
+  - name: Docker build
+    pipelineRef:
+      bundle: quay.io/redhat-appstudio-tekton-catalog/pipeline-docker-build:3649b8ca452e7f97e016310fccdfb815e4c0aa7e
+      name: docker-build
+    when:
+      dockerfile: true
+  - name: NodeJS
+    pipelineRef:
+      bundle: quay.io/redhat-appstudio-tekton-catalog/pipeline-nodejs-builder:3649b8ca452e7f97e016310fccdfb815e4c0aa7e
+      name: nodejs-builder
+    when:
+      language: nodejs,node
+  - name: Fallback
+    pipelineRef:
+      bundle: quay.io/my-user/my-fallback:v1.0
+      name: fallback
+```
+
+In the example the first selector will match only when the component CR name is `my-component` and it's defined in the devfile or detected with `language: java` and `projectType: springboot` if all requests are matching then PipelineRun is generated based on `pipelineRef` and pipelineRun parameters are set to pipelineParams. If any of `when` condition does not match then it's checking the next selector. The `Fallback` sector does not contain `when` so it will be processed in case previous selectors do not match.
```

Comment:
I am wondering .. is there a `Fallback` sector for the global BuildPipelineSelector CR`build-pipeline-selector` in `build-service` namespace in case there is not a match, how the scenario is handled and will there be any error visible to the user or suggested to define BuildPipelineSelector in the user's namespace?

Reply:
Right now it will fallback to hardcoded pipeline in build service. I added line with this informatoion.

Reply:
Thank you!